state/trackableDataTrie/trackableDataTrie.go

```diff
@@ -248,8 +248,7 @@ func (tdt *trackableDataTrie) updateTrie(dtr state.DataTrie) ([]core.TrieData, e
 }
 
 func (tdt *trackableDataTrie) retrieveValueFromTrie(key []byte) (core.TrieData, uint32, error) {
-	currentEpoch := tdt.enableEpochsHandler.GetCurrentEpoch()
-	if tdt.enableEpochsHandler.IsAutoBalanceDataTriesEnabledInEpoch(currentEpoch) {
+	if tdt.enableEpochsHandler.IsFlagEnabled(common.AutoBalanceDataTriesFlag) {
```

Comment:
I think we should add for each subcomponent also the compatibility check, to find out early that the handlers for finding that a flag is enabled are defined in the map.

Reply:
added TODO for this and I'll add the checks in the further PRs, while replacing the calls(easier to extract exactly what flags the specific subcomponent will need)